src/app/static/src/app/components/modelOutput/ModelOutput.vue

```diff
@@ -226,15 +226,23 @@
                     filters: this.barchartFilters
                 }
             },
-            flattenedFilterOptionIds(){
-                const ids: Dict<string[]> = {}
-                this.barchartFilters.forEach(filter => {
-                    if ((filter.options[0] as NestedFilterOption).children){
-                        ids[filter.id] = Object.keys(flattenOptions(filter.options))
-                    } else {
-                        ids[filter.id] = filter.options.map((option: FilterOption) => option.id)
+            flattenedXAxisFilterOptionIds(){
+                const xAxisId = this.barchartSelections?.xAxisId
+                let ids: string[] = []
+                if (xAxisId && this.barchartFilters?.length){
+                    for (let index = 0; index < this.barchartFilters.length; index++) {
+                        const filter = this.barchartFilters[index]
+                        if (filter.id === xAxisId){
+                            if ((filter.options[0] as NestedFilterOption).children){
+                                ids = Object.keys(flattenOptions(filter.options)) || []
+                                break
+                            } else {
+                                ids = filter.options.map((option: FilterOption) => option.id) || []
+                                break
+                            }
+                        }
```

Comment:
So it looks like here, you're looping through all the barchart filters in order to find the one which is the x axis and flatten that. It would be less verbose to ditch the for loop and just use find wouldn't it? Something like 
`const filter = barchartFilters.find((f: Filter) => f.id === xAxisId)`?